### DISCUSSION TITLE
Form actions return code 400 in production

### DISCUSSION THREAD

**Discussion by Hejtmus** [sveltejs/kit#13038] [Q&A]
I have deployed SvelteKit (2.8.1, but also older version share the character) app to Google Cloud Functions (which runs on Cloud Run) using node adapter, exported handler as a cloud function.

## The problem

The server correctly handles GET requests, but every non-GET request fails, in dev mode it works like expected. I have modified the adapter code, to print all incoming requests, and also placed similar logging to server hooks.
When making POST request, for instance logging in, the request reaches the first log (that means the problem is not in Cloud Run), but the request doesn't reach the server hooks.

![image](https://github.com/user-attachments/assets/9494c599-a117-4bc5-85cd-520d3c49045a)

## Fixing attempts:

- At first, I thought it's csrf, so I disabled it, same behavior.
- Next I tried setting [environmental variables](https://svelte.dev/docs/kit/adapter-node#Environment-variables-ORIGIN-PROTOCOL_HEADER-HOST_HEADER-and-PORT_HEADER), both origin, as well as requested protocol and host, logged output of get_origin, it was correct, but still the same result.
- Also I tried suggestions from [this](https://github.com/sveltejs/kit/discussions/7600) discussion, no luck.

Does anyone have idea what am I doing wrong? I can provide more info if needed.

**Comment by Nazacheres** [comment]
Hi, @Hejtmus did you manage to figure it out?

**Reply by Hejtmus** [reply]
Hey @Nazacheres ,

The problem was the adapter, more precisely its parsing http header of requests from cloud run, I wrote one for my platform.

Here you can find it: https://github.com/GenoaCMS/sveltekit-adapter-cloud-run-functions

Best regards,
Filip.

**Reply by Nazacheres** [reply]
Interestingly, it was the last straw for me with Firebase functions, and I decided to switch to Google Cloud Run yesterday; the issue disappeared automatically with the default adapter. But thanks for the link; if I decide to go back, I will look into it.
